server/src/test/scripts/data_demo.py

```diff
@@ -25,6 +25,37 @@ def demo(func):
     demos.append(func)
     return func
 
+@demo
+def demo_1d_exponential(dv):
+    dv.new('demo_1d_exponential', ['x [ms]'], ['y [V]'])
+    add_params(dv)
+
+    # Generate points between -5 and 5 in 0.1 increments
+    for i in [x * 0.1 for x in range(-50, 50)]:
+        x = i
```

Comment:
I would use `i` for the integer index and `x` for the float value:
```python
for x in (i * 0.1 for i in xrange(-50, 50)):
    ...
```
The parens and `xrange` make this a generator expression instead of precreating a list.